components/ui/AppMegaDropdownMenu.vue

```diff
@@ -88,59 +240,68 @@ export default class TheMegaDropdownMenu extends Vue {
   color: $cool-gray-80;
   position: relative;
 
-  &__button {
-    &_primary,
-    &_secondary {
+  &__search-field-wrapper {
+    display: flex;
+    width: 18rem;
+    height: 2.5rem;
+    justify-content: space-between;
+    align-items: center;
+    padding: $spacing-03;
+
+    color: $cool-gray-80;
+    fill: $cool-gray-80;
+    border-bottom: 1px solid $purple-70;
+
+    &__input {
+      flex: 1;
       background-color: transparent;
       outline: none;
       border: none;
-      width: 18rem;
-      display: flex;
-      justify-content: space-between;
-      align-items: center;
-      padding: $spacing-03;
+      color: inherit;
+    }
+    &__input:focus::placeholder {
+      opacity: 0.25;
+    }
 
-      .app-mega-dropdown__icon > path {
-        transform: translate(0, 0);
-        transition: transform 0.2s ease-in-out;
-      }
+    &__icon {
+      flex: 0 0 1rem;
+    }
 
-      &:hover {
-        cursor: pointer;
+    &__icon > path {
+      transform: translate(0, 0);
+      transition: transform 0.2s ease-in-out;
+    }
 
-        .app-mega-dropdown__icon > path {
-          transform: translate(0, 4px);
-          transition: transform 0.2s ease-in-out;
-        }
-      }
+    &:hover &__icon > path {
+      transform: translate(0, 4px);
+      transition: transform 0.2s ease-in-out;
     }
 
     &_primary {
+      background-color: transparent;
+    }
+    &_primary &__input::placeholder {
```

Comment:
We can use Sass nesting here.

Reply:
Per offline conversations, is not possible to keep code simple.